src/stateless/slider.vue

```diff
@@ -0,0 +1,546 @@
+            isDomReady: false,
+            isChanged: false,
+            isDragging: false,
+        }
+    },
+    computed: {
+        limitValue () {
+            return this.limit || this.max
+        },
+        minLabelValue () {
+            return this.minLabel || this.min.toLocaleString(this.locale)
+        },
+        maxLabelValue () {
+            return this.maxLabel || this.limitValue.toString().toLocaleString(this.locale)
+        },
+        index () {
+            return (this.value - this.min) / this.step
+        },
+        stepsCount () {
+            return (this.limitValue - this.min) / this.step
+        },
+        stepPercentage () {
+            return 1 / this.stepsCount
+        },
+        position () {
+            return this.isDomReady ? Math.max(0, Math.min(this.index, this.stepsCount)) * this.stepPercentage * this.sliderWidth : 0
+        },
+        sliderWidth () {
+            return this.isDomReady ? this.$refs.bar.clientWidth : 0
+        },
+        sliderOffset () {
+            return this.isDomReady ? this.$refs.bar.getBoundingClientRect().x : 0
+        },
+        decimalPlacesCount () {
+            let decimals = this.step.toString().split('.')[1]
+            return decimals ? decimals.length : 0
+        },
+        isWholeNumber () {
+            return this.decimalPlacesCount === 0
+        },
+        lastActiveIndex () {
+            let valueRatio = (this.value - this.min) / (this.limitValue - this.min)
+            return valueRatio * (this.ticksCount - 1)
+        },
+        stateClass () {
+            return {
+                'disabled': this.disabled,
+                'changed': this.isChanged,
+                'dragging': this.isDragging,
+            }
+        },
+        labelsClass () {
+            let isNormalSize = this.size === 'normal'
+            return {
+                min: { active: this.lastActiveIndex >= 0 && isNormalSize },
+                max: { active: this.lastActiveIndex >= 19 && isNormalSize },
+            }
+        },
+    },
+    beforeCreate () {
+        this.ticksCount = 20
+        this.labelPadding = 6
+    },
+    created () {
+        if (this.decimalPlacesCount > 1)
+            throw new Error('Max one decimal place is supported.')
+
+        let isStepValid = (this.max - this.min) / this.step % 1 === 0
+        if (!isStepValid)
+            throw new Error('Difference between max and min is not divisible by step.')
+
+        let isValueValid = ((this.value - this.min) % this.step).toFixed(0) === '0'
+        if (!isValueValid)
+            throw new Error('Value is not a valid step.')
+
+        if (this.min < 0 || this.min > 998)
+            throw new Error('Min must be between 0 and 998.')
+
+        if (this.max > 999 || this.max <= this.min)
+            throw new Error('Max must be between min+1 and 999.')
+
+        if (this.value < this.min || this.value > this.max)
+            throw new Error('Value must be between min and max.')
```

Comment:
What if this components receives invalid value after created hook? I would move this logic to a watcher.

Reply:
I know there is room for improvements in validation but let's leave it for later.